packages/ui/src/components/molecules/UserButton.tsx

```diff
@@ -0,0 +1,49 @@
+import { FC } from 'react'
+import { Avatar } from '../atoms'
+import { dimensions } from '../../styles'
+
+export type TUser = {
+    id?: string
+    name: string
+    avatarId: string
+    email?: string
+    role?: 'ADMIN' | 'REGISTERED' | 'MENTOR'
+} | null
```

Comment:
Maybe you need a type name more specific and include just what's necessary in this component, because TUser probably will be defined in the application.

Reply:
This component is just an image, does it need all this information? You just need the values for src and alt from parent if there is an user. The parent should manage if there is an user or not, and what to do if there is an onClick (open Modal or go to Profile page). 
Maybe even Avatar can serve for this purpose with a few changes...